### ISSUE TITLE
BinaryOp\SimplifyDeMorganBinaryRector -> "Class name must be a valid object or a string"

### ISSUE THREAD

**Issue by voku** [rectorphp/rector#915]
error:
```
[ERROR] Could not process "/home/lmoelleken/testing/git/portable-utf8/src/voku/helper/UTF8.php" file by                
         "Rector\CodeQuality\Rector\BinaryOp\SimplifyDeMorganBinaryRector", due to:                                     
         "Class name must be a valid object or a string".
```

info: When I delete this (from the next code example), then it works without the error message.
```php
        if (!($code & 0x80)) {
            // 0xxxxxxx
            return $code;
        }
```

code:
```php
    /**
     * Get a decimal code representation of a specific character.
     *
     * @param string $char <p>The input character.</p>
     *
     * @return int
     */
    public static function chr_to_decimal(string $char): int
    {
        $code = ord($char[0]);
        $bytes = 1;

        if (!($code & 0x80)) {
            // 0xxxxxxx
            return $code;
        }

        if (($code & 0xe0) === 0xc0) {
            // 110xxxxx
            $bytes = 2;
            $code &= ~0xc0;
        } elseif (($code & 0xf0) === 0xe0) {
            // 1110xxxx
            $bytes = 3;
            $code &= ~0xe0;
        } elseif (($code & 0xf8) === 0xf0) {
            // 11110xxx
            $bytes = 4;
            $code &= ~0xf0;
        }

        for ($i = 2; $i <= $bytes; $i++) {
            // 10xxxxxx
            $code = ($code << 6) + (ord($char[$i - 1]) & ~0x80);
        }

        return $code;
    }
```

**Comment by TomasVotruba** [comment]
Thanks for report, I'll look into it

**Comment by TomasVotruba** [comment]
Fixed in #926
